enhancements/sdk-fbc-run-bundle.md

```diff
@@ -0,0 +1,279 @@
+
+The declarative config is a struct defined by the Operator Registry that represents the index in a format that is easily transformed to a `YAML/JSON` format. The Operator SDK will invoke Operator Registry APIs to generate various components of a declarative config and eventually insert them into the declarative config object. This object will then be converted to a human-readable form like `YAML/JSON`.
+
+```go
+type DeclarativeConfig struct {
+  Packages []Package
+  Channels []Channel
+  Bundles []Bundle
+  Others []Meta
+}
+```
+
+As seen in the struct above, a declarative config has three major components - Bundles, Packages, and Channels. For more information about these subtypes, refer to [Operator Registry DeclarativeConfig definitions](<https://github.com/operator-framework/operator-registry/blob/master/alpha/declcfg/declcfg.go>).
+
+Essentially, a basic File-Based Catalog is composed of these three components in `JSON/YAML` format. The Operator SDK generates each of these components from the Operator Registry APIs in the following manner:
+
+[Bundle](<https://github.com/operator-framework/operator-sdk/blob/18ca92070fb3dd5e9f7d4581b27cc3e167a082d9/internal/olm/operator/bundle/install.go#L239>)
+
+[Package](<https://github.com/operator-framework/operator-sdk/blob/18ca92070fb3dd5e9f7d4581b27cc3e167a082d9/internal/olm/operator/bundle/install.go#L251>)
+
+[Channel](<https://github.com/operator-framework/operator-sdk/blob/18ca92070fb3dd5e9f7d4581b27cc3e167a082d9/internal/olm/operator/bundle/install.go#L260>)
+
+These three structs are used to construct the declarative config object. This declarative config object is a File-Based Catalog representation of an index with a single bundle present in it. This declarative config object can easily be transformed into `YAML/JSON` format.
+
+It is noteworthy that this enhancement proposal does not change the `run bundle` command from a user’s perspective. The command-line arguments and flags will remain the same. However, the Operator SDK will handle adding a newly-generated bundle to an index differently than it currently does. That is the goal of this enhancement proposal.
+
+#### Command Line Interface
+
+The command line interface will remain exactly the same as the original `run bundle` command.
+
+#### Assumptions
+
+Like the existing run bundle command, we **assume** the following:
+
+ 1. OLM is already installed
+ 1. kube API is available with the permissions to create resources
+ 1. kubelet can see and pull the provided images (i.e. bundle and index images)
+ 1. If an index image is provided, we assume that it has:
+    - bash, so we can change the entrypoint
+    - if it is a sqlite index image, we assume that:
+        - the existing registry DB located at `/database/index.db`
+        - dependencies required for `opm registry add` so we can inject the bundle into the DB
```

Comment:
Why the assumptions section are relevant here? Could we not break down this section into 2 descriptions?

a) If the run bundle command is called inform the bundle only:

So here we can describe how we will create a new index using the FBC format, and etc...

b) If the run bundle command is called informing an index:

So here we can describe that we will check the label to identify if the index format is SQL index or FBC format
Then, we can describe what should have in both cases. So that, we could have more 2 subsection, if the index is SQL for and if the index in the FBC format what we will describe what technically we need to change/have for it works. 

WDYT? 
Could we not simplify this doc in this way?
Would not it make more clear what changes we need to do?

Reply:
I think what you are describing fits better in the "What happens when the `run bundle` command is invoked?" section in the implementation details. I've detailed out those two scenarios there. 

This section is meant to state what assumptions we are making for `run bundle` to succeed.